### PR TITLE
Web3 Integration for WebGL

### PR DESCRIPTION
**Pull request by 0xdenkle**
### Notes for reviewer
- There's a bunch of third-party code you can skip over, just look at the Scripts folder
- Also look at the javascript component of the webgl template

### CODE REVIEW

**Comment by 0xdenkle** [comment]
Just need a review from someone else before we can merge, @drikusroor

**Review by drikusroor — APPROVED** [review]
Code looks good and well structured as far as I can see and my basic functional tests seemed successful, so I approved the MR.

I did have some nitpicks about the javascript parts, but I'll leave it up to you on whether you want to address them. My main issue was the communication. The MR had no description and there was just one commit with a gazillion files. This made it hard and tedious for me to find out your thought process and hard to find the relevant files.

But in the end I think there is some serious progress here, which is nice! 🎉

**Pull request merged**
